src/taoensso/timbre/appenders/core.cljc

```diff
@@ -82,29 +97,14 @@
 
      (have? enc/nblank-str? fname)
 
-     {:enabled? true
-      :fn
-      (let [lock (Object.)]
+     (let [lock (Object.)]
+       {:enabled? true
+        :fn
         (fn self [data]
-          (let [{:keys [output_]} data
-                output (force output_)
-                locklocal lock ; Ensure obj on stack for verifiers, Ref. CLJ-1472
-                ]
-            (try
-              (when locking? (monitor-enter locklocal)) ; For thread safety, Ref. #251
-              (with-open [^java.io.BufferedWriter w (jio/writer fname :append append?)]
-                (.write   w ^String output)
-                (.newLine w))
-
-              (catch java.io.IOException e
-                (if (:spit-appender/retry? data)
-                  (throw e) ; Unexpected error
-                  (do
-                    (jio/make-parents fname)
-                    (self (assoc data :spit-appender/retry? true)))))
-              (finally
-                (when locking?
-                  (monitor-exit locklocal)))))))}))
+          (if locking?
```

Comment:
@ericdallo If I understand @ptaoussanis correctly, the `(force output)` should happen outside of the lock, probably as the first step in the function body, outside the `if`. Correct?

Reply:
Done @ptaoussanis